Lovely edger: Placeholder avatars

whimsicallyson opened the pull request:
Placeholder avatars after you've invited someone to a team (https://www.notion.so/glitch/Show-placeholder-avatars-for-people-after-inviting-them-88fdccd9e24e4172ae9d261e83e6dc2c)

judeallred commented on src/presenters/includes/team-users.jsx:
```diff
@@ -94,6 +96,7 @@ export class AddTeamUser extends React.Component {
     this.setState({
       invitee: getDisplayName(user),
+      alreadyInvited: [...this.state.alreadyInvited, user],
     });
```
When you need to reference the current state to set the new state, we get to use the alternate setState syntax so that it can handle asynchrony -- see https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

whimsicallyson replied:
I think I updated this properly?

judeallred replied:
almost :-)  see comment on latest code

judeallred commented on src/presenters/includes/team-users.jsx:
```diff
@@ -124,6 +127,9 @@ export class AddTeamUser extends React.Component {
                 Invited {this.state.invitee}
               </div>
             }
+            {!!this.state.alreadyInvited && 
```
Empty arrays are truthy-- need to use this.state.alreadyInvited.length here if you want to prevent rendering the UsersList when the array is empty

whimsicallyson replied:
UsersList is just an empty div, so its not problematic to have it rendered, but I agree its cleaner to not.

whimsicallyson commented:
> can you add a screenshot here or in the notion card so i can get a quick glance at what this looks like?

![screen shot 2018-11-02 at 9 43 50 am](https://user-images.githubusercontent.com/3011231/47918761-dd51ae80-de83-11e8-952d-f9eef3985f50.png)

pketh commented:
In other systems with ‘shadow’ users like trello, there isn’t a strong top level distinction in position from users invited and users who’ve joined. The reason there’s no distinction is because if I’m just using my team, i can usually consider invited users already as members of the team. 

If you want to convey that user hasn’t accepted the invite yet instead of repositioning, it might be more low key to either day that in the person-pop, or display them as transparent (to hint at the ghosty ness of them)

pketh commented:
If you’d like I can move the above comment to the notion card instead, just lmk

whimsicallyson commented:
@pketh I ghosted the avatars to 50% opacity, I think that helps

judeallred commented on src/presenters/includes/team-users.jsx:
```diff
@@ -92,9 +94,10 @@ export class AddTeamUser extends React.Component {
+    this.setState((state) => ({
       invitee: getDisplayName(user),
-    });
+      alreadyInvited: [...this.state.alreadyInvited, user],
```
almost; you need to use the `state` object that comes in from the lambda rather than the `this.state` object.

whimsicallyson replied:
done!

whimsicallyson replied:
whoops, *now* its fixed.

pketh commented:
Awesometown, can you upload a new screen w the opacity?

whimsicallyson commented:
![screen shot 2018-11-02 at 1 41 15 pm](https://user-images.githubusercontent.com/3011231/47931415-1a7a6880-dea5-11e8-94fb-8b6595b11100.png)
@pketh

pketh commented:
thanks @whimsicallyson , looks great. I think because the ghost users now appear ghostly, they can appear in the normal list of users to the left of the button. 

btw what does it look like if you add a user with an unknown email? do you get the colored anon face?

whimsicallyson commented:
added support for showing added-user-by-email (whenever that ships) and removed unnecessary padding now that the avatars are to the left of the button.

Jenn says this is fine to ship without persistence as long as that part is going to be worked on soon, so @judeallred if you think we're good to go on all sides, let me know and we can merge/release!

judeallred commented:
Sounds good,  could you please extract her Jenn's persistence request out of this PR and into a notion card?  

functionally lgtm now, taking a glance at the latest code...

judeallred commented on src/presenters/includes/team-users.jsx:
```diff
@@ -92,17 +94,19 @@ export class AddTeamUser extends React.Component {
+    this.setState((state) => ({
       invitee: email,
-    });
+      alreadyInvited: [...state.alreadyInvited, email],
```
this doesn't seem right to me -- it looks like we're appending email addressses to a list of users.  I'd expect this to fail down the line when the <UsersList> receives a string email address instead of a user object.

judeallred replied:
i think it'll be hard to test this until invite-by-email (for non preexisting users) exists, so I'd suggest ignoring this part of the feature.  Speculative code is hard to write; we can add it once email invites are implemented

whimsicallyson replied:
ok, took it out 👍
